### ISSUE TITLE
-Z build-std broken on latest nightly: could not read vendor path

### ISSUE THREAD

**Issue by phil-opp** [rust-lang/rust#80148]
### Steps to reproduce

1. `cargo new --lib example && cd example`
2. `echo "#![no_std]" > src/lib.rs`
3. `cargo +nightly check --target x86_64-unknown-linux-gnu -Z build-std=core`

(The actual target doesn't seem to matter, the `build-std` feature just requires that we pass a `--target` argument.)

The following error occurs:

> error: could not read vendor path /home/philipp/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/vendor
> 
> Caused by:
>   No such file or directory (os error 2)

### Cause

I think this issue occurs because https://github.com/rust-lang/rust/pull/80082 already landed, while the cargo-side revert in https://github.com/rust-lang/rust/pull/80081 did not. So **this issue will likely go away after the next cargo update**. I still decided to open this issue in case other people encounter the same problem.

### Meta

`rustc --version --verbose`:
```
rustc 1.47.0 (18bf6b4f0 2020-10-07)
binary: rustc
commit-hash: 18bf6b4f01a6feaf7259ba7cdae58031af1b7b39
commit-date: 2020-10-07
host: x86_64-unknown-linux-gnu
release: 1.47.0
LLVM version: 11.0
```

**Comment by lzutao** [comment]
Cc @ehuss for #80082

**Comment by ehuss** [comment]
This will be fixed by #80081.
